app/controllers/search_controller.rb

```diff
@@ -1,8 +1,21 @@
 class SearchController < ApplicationController
   def search
     session[:address] = params[:search] if params.has_key?(:search)
-    @search = Listing.where(is_active: true)
-    @search = @search.near("#{session[:address]}", 5, order: 'distance')
+    if session[:address].present? && session[:address].empty? == false
+      @search = Listing.where(is_active: true)
+      if params[:check_in_date].present? && params[:check_in_date].empty? == false && params[:check_out_date].present? && params[:check_out_date].empty? == false 
+        @bookings = Booking.where("check_in_date BETWEEN '#{params[:check_in_date]}' AND '#{params[:check_out_date]}'
+                AND check_out_date BETWEEN '#{params[:check_in_date]}' AND '#{params[:check_out_date]}'")
+          .pluck(:listing_id)
+        @search = Listing.where(is_active: true).where.not(id: @bookings)
+      end
+    else
+      @bookings = Booking.where("check_in_date BETWEEN '#{params[:check_in_date]}' AND '#{params[:check_out_date]}'
+                                AND check_out_date BETWEEN '#{params[:check_in_date]}' AND '#{params[:check_out_date]}'")
+                          .pluck(:listing_id)
+      @search = Listing.where(is_active: true).where.not(id: @bookings).near("#{session[:address]}", 5, order: 'distance')
+    end
```

Comment:
This code is redundant.
Please create a simpler way to get all listings when address is empty.
Example steps:
```
if there is address
  get all Listings
else
  get Listings filtered with address
end

proceed with other filters of search method
```
You can add filter using the dates after filtering using address.